e2e/vmss.go

```diff
@@ -110,7 +110,10 @@ func addPodIPConfigsForAzureCNI(vmss *armcompute.VirtualMachineScaleSet, vmssNam
 		}
 		podIPConfigs = append(podIPConfigs, ipConfig)
 	}
-	vmssNICConfig := vmss.Properties.VirtualMachineProfile.NetworkProfile.NetworkInterfaceConfigurations[0]
+	vmssNICConfig, err := getVMSSNICConfig(vmss)
+	if err != nil {
+		return fmt.Errorf("unable to get vmss nic: %s", err)
+	}
 	ipConfigs := append(vmssNICConfig.Properties.IPConfigurations, podIPConfigs...)
```

Comment:
do you need ipConfigs? Can you just do append and assign on the same line?

Reply:
figured this was slightly more readable but can go ahead and change